buildpacks/php/tests/integration/smoke.rs

```diff
@@ -19,6 +23,62 @@ fn smoke_test_bundled_hello_world_app() {
     );
 }
 
+#[test]
+#[ignore = "integration test"]
+fn smoke_test_composer_json_scripts_as_objects() {
+    let temp = tempfile::tempdir().unwrap();
+    let source = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+        .join("tests/fixtures/smoke/hello-world")
+        .canonicalize()
+        .unwrap();
+    let app_dir = temp.path();
+    copy_dir_all(source, app_dir).unwrap();
+
+    let mut composer_json = serde_json::from_str::<serde_json::Map<_, _>>(
+        &fs::read_to_string(app_dir.join("composer.json")).unwrap(),
+    )
+    .unwrap();
+
+    composer_json.insert(
+        "scripts".to_string(),
+        json!({
+            "auto-scripts": {
+                "cache:clear": "echo 'cache:clear'",
+                "assets:install %PUBLIC_DIR%": "echo 'assets:install'",
+                "importmap:install": "echo 'importmap:install'"
+            },
+            "post-install-cmd": [
+                "@auto-scripts"
+            ],
+            "post-update-cmd": [
+                "@auto-scripts"
+            ]
+        }),
+    );
+    fs::write(
+        app_dir.join("composer.json"),
+        serde_json::to_string(&composer_json).unwrap(),
+    )
+    .unwrap();
+
+    let build_config = BuildConfig::new(builder(), app_dir)
+        .buildpacks(vec![BuildpackReference::CurrentCrate])
+        .target_triple(target_triple(builder()))
+        .to_owned();
+
+    TestRunner::default().build(&build_config, |context| {
```

Comment:
I'm not sure I quite follow why this assertion is here? I thought the bug being fixed was at build time not run time? And if it was at run time, then the `run_shell_command` doesn't actually run php?